core/deployment/src/test/java/io/quarkiverse/langchain4j/test/guardrails/InputGuardrailPromptTemplateTest.java

```diff
@@ -0,0 +1,186 @@
+package io.quarkiverse.langchain4j.test.guardrails;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+
+import jakarta.enterprise.context.RequestScoped;
+import jakarta.inject.Inject;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import dev.langchain4j.data.message.AiMessage;
+import dev.langchain4j.data.message.ChatMessage;
+import dev.langchain4j.memory.chat.ChatMemoryProvider;
+import dev.langchain4j.model.chat.ChatLanguageModel;
+import dev.langchain4j.model.output.Response;
+import dev.langchain4j.service.MemoryId;
+import dev.langchain4j.service.UserMessage;
+import io.quarkiverse.langchain4j.RegisterAiService;
+import io.quarkiverse.langchain4j.guardrails.InputGuardrail;
+import io.quarkiverse.langchain4j.guardrails.InputGuardrailResult;
+import io.quarkiverse.langchain4j.guardrails.InputGuardrails;
+import io.quarkiverse.langchain4j.runtime.aiservice.NoopChatMemory;
+import io.quarkus.arc.Arc;
+import io.quarkus.test.QuarkusUnitTest;
+
+public class InputGuardrailPromptTemplateTest {
+
+    @RegisterExtension
+    static final QuarkusUnitTest unitTest = new QuarkusUnitTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addClasses(MyAiService.class, MyAiService.class, GuardrailValidation.class,
+                            MyChatModel.class, MyChatModelSupplier.class, MyMemoryProviderSupplier.class));
+    @Inject
+    MyAiService aiService;
+
+    @Inject
+    GuardrailValidation guardrailValidation;
+
+    @BeforeEach
+    void setup() {
+        Arc.container().requestContext().activate();
+    }
+
+    @AfterEach
+    void tearDown() {
+        Arc.container().requestContext().deactivate();
+    }
+
+    @Test
+    void shouldWorkNoParameters() {
+        aiService.getJoke();
+        assertThat(guardrailValidation.spyPromptTemplate()).isEqualTo("Tell me a joke");
+        assertThat(guardrailValidation.spyVariables()).isEmpty();
+    }
+
+    @Test
+    void shouldWorkWithMemoryId() {
+        aiService.getAnotherJoke("memory-id-001");
+        assertThat(guardrailValidation.spyPromptTemplate()).isEqualTo("Tell me another joke");
+        assertThat(guardrailValidation.spyVariables()).containsExactlyInAnyOrderEntriesOf(Map.of(
+                "memoryId", "memory-id-001",
```

Comment:
I believe the memory id should not be part of the variable, but deserve its own method. @geoand WDYT?

Reply:
Agreed

Reply:
I have found [this](https://github.com/quarkiverse/quarkus-langchain4j/blob/fa11843bba5aa1ead0b3fb504bfb4b24eee53902/core/deployment/src/test/java/io/quarkiverse/langchain4j/test/AiServiceMethodParametersAnnotationsTest.java#L149) test that checks if the memory id is in the template.

So, looks like it was designed to be like that, I need to know if we will remove it or keep it before continuing...

Reply:
Ok, maybe we can keep it. The "name" should be the parameter name.